### PR TITLE
adding VStringCodeField, VStringField

### PR DESCRIPTION
**Pull request by h-haddad**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt [hunk 1]
```diff
@@ -17,4 +17,283 @@
...
+import org.kopi.galite.list.VStringColumn
+import org.kopi.galite.util.LineBreaker
+import org.kopi.galite.visual.VExecFailedException
+import org.kopi.galite.visual.VlibProperties
+import org.kopi.galite.base.Query
+
+open class VStringField(width: Int, height: Int, private val visibleHeight: Int, val typeOptions: Int, styled: Boolean)
+  : VField(width, height) {
+
+  constructor(width: Int, height: Int, convert: Int, styled: Boolean) : this(width, height, 0, convert, styled) {}
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt:33]: Remove empty {}

File: galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt [hunk 2]
```diff
@@ -17,4 +17,283 @@
...
+  override fun getListColumn(): VListColumn {
+    return VStringColumn(getHeader(), null, align, width, getPriority() >= 0)
+  }
+
+  /**
+   * verify that text is valid (during typing)
+   */
+  override fun checkText(s: String): Boolean {
+    var end = 0
+    end = textToModel(s, width, Int.MAX_VALUE).length
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt:82]: Empty line

File: galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt [hunk 3]
```diff
@@ -17,4 +17,283 @@
...
+
+  /**
+   * Returns the field value of the current record as an object
+   */
+  override fun getObjectImpl(r: Int): Any? {
+    return value[r]
+  }
+
+  override fun toText(o: Any?): String {
+    // TODO Auto-generated method stub
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt:195]: is it for todo or is it not?

**Review by ZMarouani — CHANGES_REQUESTED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt [hunk 4]
```diff
@@ -17,4 +17,282 @@
...
+  /**
+   * Returns the field value of the current record as an object
+   */
+  override fun getObjectImpl(r: Int): Any? {
+    return value[r]
+  }
+
+  override fun toText(o: Any?): String {
+    return if (o == null) "" else (o as String?)!!
+  }
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt:196]: = (o as String?).orEmpty()

File: galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt [hunk 5]
```diff
@@ -17,4 +17,282 @@
...
+    fun textToModel(source: String, col: Int): String {
+      return textToModel(source, col, Int.MAX_VALUE)
+    }
+    /**
+     * Replaces new-lines by blanks
+     *
+     * @param    source    the source text with carriage return
+     * @param    col    the width of the text
+     */
+
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt:281]: Del

**Review by ZMarouani — CHANGES_REQUESTED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt [hunk 6]
```diff
@@ -17,4 +17,282 @@
...
+    for (i in chars.indices) {
+      if (!found && Character.isLetter(chars[i])) {
+        chars[i] = Character.toUpperCase(chars[i])
+        found = true
+      } else if (Character.isWhitespace(chars[i])) {
+        found = false
+      }
+    }
+    return String(chars)
+  }
```
[the remark this answers is not in the document]
Comment (h-haddad) [reply galite-core/src/main/kotlin/org/kopi/galite/form/VStringField.kt:128]:
```
private fun convertName(source: String): String {
    val chars = source.toLowerCase().toCharArray()
    var found = false

    for (i in chars.indices) {
      if (!found && chars[i].isLetter()) {
        chars[i] = chars[i].toUpperCase()
        found = true
      } else if (chars[i].isWhitespace()) {
        found = false
      }
    }
    return String(chars)
  }
```

**Review by ZMarouani — APPROVED** [review]

**Pull request merged**